### PR TITLE
General updates mainly to integrate MUSE better: integrate more methods, model, improve code structure, set hyperparams for MUSE exps and misc fixes. 

### PR DESCRIPTION
**Pull request by Dornavineeth**
# Done
- [x] Fix trainer.evaluate
- [x] Fix OOM issue of methods using reference model.
- [x] Fix batch_size in evaluation -> No OOM in eval during training/unlearning
- [x] SimNPO added
- [x] Default learning rate of tofu unlearning from 5e-5 to 2e-5
- [x] asst_start_tag="" for llama2-7b-chat to match official TOFU implementation
- [x] Update way of computing label_ids at the boundary of prompt and response - corner case

# Unlearning baselines supported
- [x] Gradient Ascent
- [x] GradDiff
- [x] NPO
- [x] IDK DPO
- [x] SimNPO
- [ ] RMU
- [ ] SOUL
- [ ] Task Vectors

# TODO
- [ ] Repro privleak of MUSE on target model.
- [ ] Tune SimNPO to reproduce reasonable results.
- [ ] Update documentations.

### CODE REVIEW

File: configs/model/Llama-2-7b-chat-hf.yaml
Comment (molereddy) [inline configs/model/Llama-2-7b-chat-hf.yaml:1]: imo- we should not change this for llama2 in general it should only be for the tofu version, either have a separate model for tofu or override in experimental configs for tofu

File: configs/trainer/finetune.yaml [hunk 1]
```diff
@@ -13,9 +13,10 @@ args:
   optim: paged_adamw_32bit
   save_strategy: 'no'
   save_only_model: True
-  weight_decay: 0.01
+  weight_decay: 0.00
```
Comment (molereddy) [inline configs/trainer/finetune.yaml:16]: why change this?

File: src/trainer/unlearn/simnpo.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        super().__init__(*args, **kwargs)
+        self.delta = delta
+        self.beta = beta
+
+    def compute_loss(self, model, inputs, return_outputs=False):
+        forget_inputs = inputs["forget"]
+
+        forget_labels = forget_inputs["labels"]
+        loss_mask = forget_labels != -100
+        forget_loss, forget_outputs = forget_outputs = compute_batch_nll(
```
Comment (molereddy) [inline src/trainer/unlearn/simnpo.py:18]: fix this line

Comment (molereddy) [reply src/trainer/unlearn/simnpo.py:18]: is this code tested, btw?

File: src/evals/metrics/privacy.py
Comment (molereddy) [inline src/evals/metrics/privacy.py:1]: raise error instead of setting p-value to 1. one can't assume no retain logs => retain model training

File: configs/trainer/finetune.yaml (hunk 1 above)
Comment (Dornavineeth) [reply configs/trainer/finetune.yaml:16]: MUSE uses default hf weight decay of 0.0 and tofu sets it to 0.01
So better have this 0.0 and in tofu experiment configs set weight_decay=0.01

File: notebooks/Analysis.ipynb
Comment (molereddy) [inline notebooks/Analysis.ipynb:1]: gitignore this

File: notebooks/plots/.gitignore
Comment (molereddy) [inline notebooks/plots/.gitignore:1]: gitignore the whole notebooks directory
